### ISSUE TITLE
Metric accumulation for AimLogger

### ISSUE THREAD

**Issue by mikel-brostrom** [aimhubio/aim#1287]
## 🚀 Feature

AimLogger should have an accumulate argument in the same way that 

- the built-in PytorchLighning logger has `on_epoch`: https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#log-dict or
- Torchmetrics has `torchmetrics.Accuracy()`: https://torchmetrics.readthedocs.io/en/latest/

### Motivation

Not need to accumulated programmatically before logging

### Pitch

I want to be able to call
`self.log('test_loss', loss, on_epoch=True)`

such that AIM only logs when the epoch is done

**Comment by alberttorosyan** [comment]
Hi @mikel-brostrom!
Thanks for opening this enhancement request. Is there any additional context you might share?
A small usage example would help greatly.

**Comment by mikel-brostrom** [comment]
In pytorch lightning, with the tensorboard logger, it would look like this:

```python
class LitClassifier(pl.LightningModule):
    def __init__(self, hidden_dim=128, learning_rate=1e-3):
        super().__init__()
        self.save_hyperparameters()

        self.l1 = torch.nn.Linear(28 * 28, self.hparams.hidden_dim)
        self.l2 = torch.nn.Linear(self.hparams.hidden_dim, 10)

    def forward(self, x):
        x = x.view(x.size(0), -1)
        x = torch.relu(self.l1(x))
        x = torch.relu(self.l2(x))
        return x

    def training_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        self.log('train_loss', loss, on_epoch=True)  # <---------- (automatically accumulates and logs at the end of the epoch)
        return loss
```

with aim as logger this is not an option

```python
class LitClassifier(pl.LightningModule):
    def __init__(self, hidden_dim=128, learning_rate=1e-3):
        super().__init__()
        self.save_hyperparameters()

        self.l1 = torch.nn.Linear(28 * 28, self.hparams.hidden_dim)
        self.l2 = torch.nn.Linear(self.hparams.hidden_dim, 10)

    def forward(self, x):
        x = x.view(x.size(0), -1)
        x = torch.relu(self.l1(x))
        x = torch.relu(self.l2(x))
        return x

    def training_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        self.log('train_loss', loss)  # <---------- (logs for each batch)
        return loss
```

**Comment by mihran113** [comment]
@mikel-brostrom I've just tested this, and it works fine with `AimLogger` as well, as the `log()` method is handled by `LightningModule` itself. I've used https://github.com/aimhubio/aim/blob/main/examples/pytorch_lightning_track.py this example and added `on_epoch=True` for `log()` method calls and everything works as expected.

**Comment by mikel-brostrom** [comment]
Yes, you are right. This works for log but does not seem to work for log_dict. You can reproduce the error by the following code snippet

```python
from aim.pytorch_lightning import AimLogger

from argparse import ArgumentParser

import torch
import pytorch_lightning as pl
from torch.nn import functional as F
from torch.utils.data import DataLoader, random_split

try:
    from torchvision.datasets.mnist import MNIST
    from torchvision import transforms
except Exception as e:
    from tests.base.datasets import MNIST


class LitClassifier(pl.LightningModule):
    def __init__(self, hidden_dim=128, learning_rate=1e-3):
        super().__init__()
        self.save_hyperparameters()

        self.l1 = torch.nn.Linear(28 * 28, self.hparams.hidden_dim)
        self.l2 = torch.nn.Linear(self.hparams.hidden_dim, 10)

    def forward(self, x):
        x = x.view(x.size(0), -1)
        x = torch.relu(self.l1(x))
        x = torch.relu(self.l2(x))
        return x

    def training_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        metrics = {'train_loss': loss, 'some_other_metric': 3}
        self.log_dict(metrics, self.current_epoch)
        return loss

    def validation_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        self.log('val_loss', loss)

    def test_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        
        self.log('test_loss', loss)
        # Track metrics manually
        self.logger.experiment.track(1, name='manually_tracked_metric')

    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters(), lr=self.hparams.learning_rate)

    @staticmethod
    def add_model_specific_args(parent_parser):
        parser = ArgumentParser(parents=[parent_parser], add_help=False)
        parser.add_argument('--hidden_dim', type=int, default=128)
        parser.add_argument('--learning_rate', type=float, default=0.0001)
        return parser


def cli_main():
    pl.seed_everything(1234)

    # ------------
    # args
    # ------------
    parser = ArgumentParser()
    parser.add_argument('--batch_size', default=32, type=int)
    parser = pl.Trainer.add_argparse_args(parser)
    parser = LitClassifier.add_model_specific_args(parser)
    args = parser.parse_args()

    # ------------
    # data
    # ------------
    dataset = MNIST('', train=True, download=True, transform=transforms.ToTensor())
    mnist_test = MNIST('', train=False, download=True, transform=transforms.ToTensor())
    mnist_train, mnist_val = random_split(dataset, [55000, 5000])

    train_loader = DataLoader(mnist_train, batch_size=args.batch_size)
    val_loader = DataLoader(mnist_val, batch_size=args.batch_size)
    test_loader = DataLoader(mnist_test, batch_size=args.batch_size)

    # ------------
    # model
    # ------------
    model = LitClassifier(args.hidden_dim, args.learning_rate)

    # ------------
    # training
    # ------------
    aim_logger = AimLogger(
        experiment='pt_lightning_exp',
        train_metric_prefix='train_',
        test_metric_prefix='test_',
        val_metric_prefix='val_',
    )
    # log remotely on aim server
    aim_logger = AimLogger(
        repo='0.0.0.0:53800',
        experiment='blub',
        train_metric_prefix='train_',
        test_metric_prefix='test_',
        val_metric_prefix='val_',
    )
    trainer = pl.Trainer(logger=aim_logger)
    trainer.fit(model, train_loader, val_loader)

    # ------------
    # testing
    # ------------
    trainer.test(test_dataloaders=test_loader)


if __name__ == '__main__':
    cli_main()
```

I get the following error:

```bash
raise MisconfigurationException(
pytorch_lightning.utilities.exceptions.MisconfigurationException: You called `self.log(train_loss, ...)` twice in `training_step` with different arguments. This is not allowed
```

**Comment by mikel-brostrom** [comment]
Yes, you are right. `log_dict` works as expected as well. The error was on my side
